src/test/java/org/sagebionetworks/bridge/spring/controllers/SharedAssessmentControllerTest.java

```diff
@@ -0,0 +1,205 @@
+        assertEquals(dto.getGuid(), GUID);
+    }
+    
+    @Test
+    public void getLatestSharedAssessment() {
+        Assessment assessment = AssessmentTest.createAssessment();
+        when(mockService.getLatestAssessment(SHARED_STUDY_ID_STRING, IDENTIFIER)).thenReturn(assessment);
+        
+        AssessmentDto dto = controller.getLatestSharedAssessment(IDENTIFIER);
+        assertEquals(dto.getIdentifier(), IDENTIFIER);
+    }
+
+    @Test
+    public void getSharedAssessmentById() {
+        Assessment assessment = AssessmentTest.createAssessment();
+        when(mockService.getAssessmentById(SHARED_STUDY_ID_STRING, IDENTIFIER, 10)).thenReturn(assessment);
+        
+        AssessmentDto dto = controller.getSharedAssessmentById(IDENTIFIER, "10");
+        assertEquals(dto.getGuid(), GUID);
+    }
+
+    @Test
+    public void getSharedAssessmentRevisionsByGuid() {
+        PagedResourceList<Assessment> page = new PagedResourceList<>(ImmutableList.of(mockAssessment), 100);
+        when(mockService.getAssessmentRevisionsByGuid(
+                SHARED_STUDY_ID_STRING, GUID, 10, 25, true)).thenReturn(page);
+        
+        PagedResourceList<AssessmentDto> retValue = controller.getSharedAssessmentRevisionsByGuid(GUID, "10", "25", "true");
+        assertEquals(retValue.getItems().size(), 1);
+    }
+
+    @Test
+    public void getSharedAssessmentRevisionsById() {
+        PagedResourceList<Assessment> page = new PagedResourceList<>(ImmutableList.of(mockAssessment), 100);        
+        when(mockService.getAssessmentRevisionsById(
+                SHARED_STUDY_ID_STRING, IDENTIFIER, 0, API_DEFAULT_PAGE_SIZE, false)).thenReturn(page);
+        
+        PagedResourceList<AssessmentDto> retValue = controller.getSharedAssessmentRevisionsById(IDENTIFIER, null, null, "false");
+        assertEquals(retValue.getItems().size(), 1);
+    }
+
+    @Test
+    public void updateSharedAssessment() throws Exception {
+        // You do need a session for this call
+        UserSession session = new UserSession();
+        session.setStudyIdentifier(TEST_STUDY);
+        doReturn(session).when(controller).getAuthenticatedSession(DEVELOPER);
+        
+        Assessment assessment = AssessmentTest.createAssessment();
+        when(mockService.updateSharedAssessment(eq(TEST_STUDY_IDENTIFIER), any())).thenReturn(assessment);
+        
+        mockRequestBody(mockRequest, AssessmentDto.create(assessment));
+
+        AssessmentDto retValue = controller.updateSharedAssessment(GUID);
+        assertEquals(retValue.getIdentifier(), assessment.getIdentifier());
+        assertEquals(retValue.getGuid(), assessment.getGuid());
+        
+        verify(mockService).updateSharedAssessment(eq(TEST_STUDY_IDENTIFIER), assessmentCaptor.capture());
+        Assessment captured = assessmentCaptor.getValue();
+        assertEquals(captured.getIdentifier(), assessment.getIdentifier());
+        assertEquals(captured.getGuid(), assessment.getGuid());
```

Comment:
Check appId?

Reply:
No longer has this value in the object, it is passed via the method calls, and this is verified.